### DISCUSSION TITLE
Capture Client IP or Real IP when using AWS LB

### DISCUSSION THREAD

**Discussion by prawen** [nginx/kubernetes-ingress#6214] [Q&A]
Hello,

I'm trying to capture and pass the Client IP or Real IP to the upstream. Below is my deployment model.

Client -> AWS ALB -> Nginx-Ingress deployed on self managed K8s Cluster (1.30) on AWS EC2

I tried using snippet

```
    nginx.org/location-snippets: |
      proxy_set_header X-ABC-CLIENTIP $proxy_add_x_forwarded_for;
```

But it is not capturing the real IP, rather giving the private IP. 
Also, the header case is not retained. It is shown as `X-Abc-Clientip`

AWS ALB gives real ip using X-Forwarded-For it seems. May I know how we can configure to use that header as is and get the real IP ? 

Does anyone has experience on similar setup ? Any help is appreciated. Thanks.

**Comment by j1m-ryan** [comment]
Hi @prawen, I will close this so we keep the discussion in https://github.com/nginxinc/kubernetes-ingress/issues/6228
